src/components/overlay/index.js

```diff
@@ -44,22 +48,32 @@ export default class Overlay extends PureBaseComponent {
     }
   }
 
+  renderCustomContent = () => {
+    const {customContent} = this.props;
+    return <View style={styles.customContent}>{customContent()}</View>;
+  };
+
   renderImage(typeStyle) {
-    const {type} = this.props;
+    const {type, customContent} = this.props;
     const image = type !== OVERLY_TYPES.SOLID ? gradientImage : undefined;
 
-    return <Image style={[styles.container, typeStyle]} resizeMode={'stretch'} source={image}/>;
+    return (
+      <View style={styles.container}>
+        {type && <Image style={[styles.container, typeStyle]} resizeMode={'stretch'} source={image}/>}
+        {customContent && this.renderCustomContent()}
+      </View>
+    );
   }
 
   render() {
-    const {type} = this.props;
+    const {type, customContent} = this.props;
 
     if (type === OVERLY_TYPES.VERTICAL) {
-
       return (
         <View style={[styles.container]}>
           {this.renderImage(styles.top)}
           {this.renderImage(styles.bottom)}
+          {customContent && this.renderCustomContent()}
```

Comment:
you render the custom content too many times, in this example 
```
<Image
          customOverlay={() => <View><Text>A</Text></View> }
          overlayType={'vertical'}
          style={{width: 100, height: 100}}
          source={{
            uri:
              'https://images.pexels.com/photos/3326362/pexels-photo-3326362.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500'
          }}
        />
```

it will be rendered 3 times.

Reply:
ooooooh weeeeee...